### PR TITLE
add zero-address checks

### PR DESCRIPTION
**Pull request by kamescg**
https://github.com/code-423n4/2021-06-pooltogether-findings/issues/65

### CODE REVIEW

File: contracts/IdleYieldSource.sol [hunk 1]
```diff
@@ -56,7 +56,7 @@ contract IdleYieldSource is IProtocolYieldSource, Initializable, ReentrancyGuard
     function initialize(
         address _idleToken
     ) public initializer {
-
+        require(address(_idleToken) != address(0), "IdleYieldSource/idleToken-not-zero");
```
Comment (PierrickGT) [inline contracts/IdleYieldSource.sol]: We should also verify that the address passed is a contract. For this, we can use the util function `isContract` from OpenZeppelin, it will also check for the zero address.
https://docs.openzeppelin.com/contracts/4.x/api/utils#Address-isContract-address-

Add also some unit tests to test that we can't initialize if we pass address zero or a random address that is not a contract.

**Review by PierrickGT — CHANGES_REQUESTED** [review]

File: contracts/IdleYieldSource.sol (hunk 1 above)
Comment (PierrickGT) [reply contracts/IdleYieldSource.sol]: After talking with Brendan, we decided that only checking for address zero is enough. That being said the error message could be a bit more explicit. 
```suggestion
        require(address(_idleToken) != address(0), "IdleYieldSource/idleToken-not-zero-address");
```

**Review by PierrickGT — APPROVED** [review]
